marimo/_plugins/ui/_impl/table.py

```diff
@@ -322,25 +324,57 @@ def __init__(
         # (searching operations include query, sort, filter, etc.)
         self._searched_manager = self._manager
         # Holds the data after user selecting from the component
-        self._selected_manager: Optional[TableManager[Any]] = None
+        self._selected_manager: Optional[
+            Union[TableManager[Any], list[TableCell]]
+        ] = None
 
         self._selection = selection
         self._has_any_selection = False
         initial_value = []
         if initial_selection and self._manager.supports_selection():
-            if selection == "single" and len(initial_selection) > 1:
+            if (selection in ["single", "single-cell"]) and len(
+                initial_selection
+            ) > 1:
                 raise ValueError(
                     "For single selection mode, initial_selection can only contain one row index"
                 )
             try:
-                self._selected_manager = self._searched_manager.select_rows(
-                    initial_selection
-                )
+                if selection in ["single-cell", "multi-cell"]:
+                    self._selected_manager = (
+                        self._searched_manager.select_cells(
+                            [
+                                TableCoordinate(
+                                    v[0],
+                                    v[1],
+                                )
+                                for v in initial_selection
+                                if isinstance(v, tuple)
+                            ]
+                        )
+                    )
+                else:
+                    self._selected_manager = (
+                        self._searched_manager.select_rows(
+                            [
+                                v
+                                for v in initial_selection
+                                if isinstance(v, int)
+                            ]
+                        )
+                    )
             except IndexError as e:
                 raise IndexError(
                     "initial_selection contains invalid row indices"
                 ) from e
-            initial_value = initial_selection
+
+            for v in initial_selection:
```

Comment:
@mscolnick how do I make py-check happy here?

Reply:
you could either case or `# type: ignore[reason]`